### PR TITLE
chore: add nonfunctioning pseudocolumn versioning to source steps

### PR DESCRIPTION
**Pull request by Sullivan-Patrick**
### Description 
Adding psuedocolumn field to `SourceStep`s as part of [KLIP-50](https://github.com/confluentinc/ksql/blob/master/design-proposals/klip-50-partition-and-offset-in-ksqldb.md). With this PR, the pseudocolumns don't get written into the command topic, therefore are "nonfunctioning".  In future PRs, these will map to pseudocolumns corresponding to the pseudocolumn version number of a specific query.

### Testing done 
No testing on account of difficulties surrounding writing unit tests for this. Currently it seems like our current tests should handle this- a written versionNumber makes the entire server degrade and is fairly obvious- but regardless, I'm open to ideas if anyone thinks of a way to test this granularly.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-engine/src/main/java/io/confluent/ksql/execution/json/PlanJsonMapper.java
Comment (Sullivan-Patrick) [inline ksqldb-engine/src/main/java/io/confluent/ksql/execution/json/PlanJsonMapper.java:44]: Anyone who has input on if this is a reasonable thing to do, please chime in. It's necessary to make this work with an int as versionNumbers, but if there is a good reason to, we can use Optionals instead.

**Review by vcrfxia — COMMENTED** [review]
Thanks @Sullivan-Patrick -- comments inline. I've also updated the PR title from `feat` to `chore` since this PR is a non-functional change. (Another good litmus test when deciding which commit type to use is to ask whether it's useful for ksqlDB users for this PR to show up in the changelog or not. If the answer is no, then `feat` is not the right commit type for the PR.)

File: ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/SourceStep.java
Comment (Sullivan-Patrick) [reply ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/SourceStep.java:37]: My excuse is that both of these issues are remnants of playing around with Jackson! Originally this had an `JsonIgnore` above it and was given a default value of 0. Thanks for the catch.

File: ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/SourceStep.java
Comment (vcrfxia) [inline ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/SourceStep.java:36]: Can we rename this to `pseudocolumnVersion`? The current name is misleading since it makes it sound like we're versioning the step itself, but really we're just versioning the pseudocolumns.

File: ksqldb-engine/src/main/java/io/confluent/ksql/execution/json/PlanJsonMapper.java
Comment (vcrfxia) [reply ksqldb-engine/src/main/java/io/confluent/ksql/execution/json/PlanJsonMapper.java:44]: If we go the route of disabling this mapper property, let's just remove the line (in order to have the property disabled by default) rather than explicitly disabling, in order to maintain consistency with the other mappers that also disable the property.

**Review by vcrfxia — COMMENTED** [review]
Thanks @Sullivan-Patrick -- almost there! Would be good to get @rodesai 's thoughts on removing `FAIL_ON_NULL_FOR_PRIMITIVES` from PlanJsonMapper versus using Optional for the new pseudocolumn version field since I'm not clear on the original motivation for enabling `FAIL_ON_NULL_FOR_PRIMITIVES`.

File: ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/SourceStep.java
Comment (Sullivan-Patrick) [reply ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/SourceStep.java:36]: Ok, I'll wait on his feedback before merging. I've made the other changes locally.

File: ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java
Comment (Sullivan-Patrick) [inline ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java:35]: arbitrary number since nothing is written

File: ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/StreamSource.java
Comment (vcrfxia) [inline ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/StreamSource.java:54]: Let's pull the value 0 out into a constant, e.g., `LEGACY_PSEUDOCOLUMN_VERSION_NUMBER`.

File: ksqldb-execution/src/test/java/io/confluent/ksql/execution/plan/StreamSourceTest.java
Comment (vcrfxia) [inline ksqldb-execution/src/test/java/io/confluent/ksql/execution/plan/StreamSourceTest.java:54]: `OptionalInt.of(0)` and `OptionalInt.empty()` are expected to compare the same, right? If so, it'd be nice to add a case for this into this unit test (and similarly for the other classes too).

**Review by vcrfxia — COMMENTED** [review]
LGTM besides the two minor comments inline!

**Review by vcrfxia — APPROVED** [review]
LGTM! Will merge once the build passes.

**Pull request merged**
